tests/rptest/tests/many_partitions_test.py

```diff
@@ -192,13 +190,14 @@ def test_many_partitions(self, n_partitions: int, n_topics: int):
         total_memory = psutil.virtual_memory().total
         self.logger.info(f"Memory: {total_memory}")
         if total_memory < resource_settings.memory_mb * 1024 * 1024 * 3:
-            if os.environ.get('CI', None) == 'false':
+            if self.ci_mode:
+                raise RuntimeError(
+                    f"Too little memory {total_memory} on test machine")
+            else:
                 self.logger.warn(
                     f"Skipping resource intensive test, running on low-memory machine with {total_memory} bytes of RAM"
                 )
-            else:
-                raise RuntimeError(
-                    f"Too little memory {total_memory} on test machine")
+                return
```

Comment:
does this mean that when running outside of build kite (either local dev machine or the upcoming cluster mode) that CI=true should be set so that we don't silently skip this test (as opposed to getting the exception to treat low-memory node as a test failure)?

Reply:
On local dev machine, it's intended to silently (well, silently but with a log warning) skip the test.

In the cluster mode, I would hope we'll be running those in CI at some stage too, but it might also be useful to have a variable that conveys the "not a laptop" condition, which is what I'm really trying to test here.

Reply:
ah yeh ok that makes sense